app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/name/SubscribedWithoutIdChangeContactNameController.scala

```diff
@@ -0,0 +1,87 @@
+
+import cats.data.EitherT
+import cats.syntax.eq._
+import cats.instances.future._
+import com.google.inject.{Inject, Singleton}
+import play.api.mvc.{Call, MessagesControllerComponents, Result}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.config.{ErrorHandler, ViewConfig}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.SessionUpdates
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.actions.{AuthenticatedAction, RequestWithSessionData, SessionDataAction, WithAuthAndSessionDataAction}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.JourneyStatus.Subscribed
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.name.{ContactName, IndividualName}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.{Error, SessionData}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.SubscriptionService
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.Logging
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.{controllers, views}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SubscribedWithoutIdChangeContactNameController @Inject()(
+  val authenticatedAction: AuthenticatedAction,
+  val sessionDataAction: SessionDataAction,
+  val subscriptionService: SubscriptionService,
+  cc: MessagesControllerComponents,
+  val sessionStore: SessionStore,
+  val errorHandler: ErrorHandler,
+  val enterNamePage: views.html.name.enter_name
+)(implicit val viewConfig: ViewConfig, val ec: ExecutionContext)
+    extends FrontendController(cc)
+    with WithAuthAndSessionDataAction
+    with SessionUpdates
+    with Logging
+    with IndividualNameController[Subscribed] {
+
+  override def validJourney(request: RequestWithSessionData[_]): Either[Result, (SessionData, Subscribed)] =
+    request.sessionData.flatMap(s => s.journeyStatus.map(s -> _)) match {
+      case Some((sessionData, r: Subscribed)) => Right(sessionData -> r)
+      case _                                  => Left(Redirect(controllers.routes.StartController.start()))
+    }
+
+  override def updateName(journey: Subscribed, name: IndividualName)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, Subscribed] = {
+    val contactName = s"${name.firstName} ${name.lastName}"
+    val journeyWithUpdatedName =
+      journey.subscribedDetails.copy(name = Right(name), contactName = ContactName(contactName))
+
+    if (journey.subscribedDetails === journeyWithUpdatedName)
+      EitherT.rightT[Future, Error](journey)
+    else
+      subscriptionService
+        .updateSubscribedDetails(journeyWithUpdatedName)
+        .map(_ => journey.copy(journeyWithUpdatedName))
+
+  }
+
+  override def name(journey: Subscribed): Option[IndividualName] = {
+    val Array(first, last) = journey.subscribedDetails.contactName.value.split(" ")
```

Comment:
is this safe or even correct?

Reply:
you should be able to set this to `None` - all this is used for is to pre-populate the form with a name if one already exists which I don't think we need here

Reply:
kl

Reply:
but if we did want to prepopulate it we'd use the `journey.name` field instead of the `contactName` field here and look for a `Right`

Reply:
replaced it with 

```scala
  override def name(journey: Subscribed): Option[IndividualName] = journey.subscribedDetails.name match {
    case Left(_)               => None
    case Right(individualName) => Some(individualName)
  }
```